src/Beacon.jsx

```diff
@@ -185,12 +204,38 @@ export class Beacon extends Component {
     }
   }
 
+  clampValueToRange(val, min, max) {
+    return Math.min(Math.max(min, val), max);
+  }
+
+  getArrowHorizontalPosition() {
+    const targetBounds = this.state.targetBounds;
+    const tooltipBounds = this.state.tooltipBounds;
+    const arrowBounds = this.state.arrowBounds;
+    const targetCenter = targetBounds.left + (targetBounds.width / 2);
+
+    // Account for animation scaling/moving the elements before it happens
+    let tooltipLeft = tooltipBounds.left;
+    if (this.state.tooltipAttachmentHorizontal === 'left') {
+      tooltipLeft = tooltipLeft + ANIMATION_MOTION_STEP/2;
+    } else if (this.state.tooltipAttachmentHorizontal === 'right') {
+      tooltipLeft = tooltipLeft - ANIMATION_MOTION_STEP/2;
+    }
+
+    // Calculate real distance from target center and adjust to range
+    const edgeDistance = targetCenter - tooltipLeft;
+    const tooltipSize = (tooltipBounds.width / (ANIMATION_MOTION_SCALE * 100)) * 100;
+    const arrowSize = (arrowBounds.width / (ANIMATION_MOTION_SCALE * 100)) * 100;
+    const arrowMaxLeft = tooltipSize - ARROW_MIN_POSITION - arrowSize;
+    const arrowPos = this.clampValueToRange(edgeDistance, ARROW_MIN_POSITION, arrowMaxLeft);
+
+    return arrowPos;
+  }
```

Comment:
Vertical position is always 50%, so we only need to calculate the cases where arrow is moved on the horizontal axis.